### ISSUE TITLE
Option to rescan sound devices in settings

### ISSUE THREAD

**Issue by netnazgul** [OpenRA/OpenRA#17651]
OpenRA (as far as I'm aware) only scans for available sound outputs once during app launch. If sound configuration is changed (devices removed/added, OS sound output changed), OpenRA loses sound until restarted.

**Comment by netnazgul** [comment]
Discord log related to the issue

> The Navigator (Bain):
> Am I the only person that has huge sound issue with openRA
> 
> The Navigator (Bain):
> for exmaple
> 
> The Navigator (Bain):
> if my wireless headphones lose connection for a second
> 
> The Navigator (Bain):
> I have to restart the game to get game sound back
> 
> The Navigator (Bain):
> and if the game does not connect to the device i'm using, for when reboting it will crash
> 
> The Navigator (Bain):
> I've had this issue for a long time, on a laptop and on a pc.
> 
> The Navigator (Bain):
> using different headphones and earphones had still the same issue .
> 
> penev92:
> the first one is simple - when your headphones lose connection the OS switches the output device, but OpenRA doesn't register that and keeps trying to use the old one
> 
> penev92:
> the second one I didn't understand
> 
> The Navigator (Bain):
> no i'm only on one output
> 
> The Navigator (Bain):
> i can't get sound at all
> 
> The Navigator (Bain):
> perhaps I should check this again when it happend, but as far as i'm aware, it won't go back to the headphones
> 
> WhoCares:
> if I plug/unplug my headphone or, as @penev92 stated, make a change in the output device, openra loses sound and i have to restart the game. always has been.
> 
> netnazgul:
> afair OpenRA only scans for sound devices once on launch. I also had it very annoying to not picking up/not being able to change sound input correctly when I plugged in/out my USB headphones.

**Comment by pchote** [comment]
Somewhat related to #12300, which we fixed in the hotfix. Fixing this goes well with the theme of the settings menu overhaul, so I'll take this on for Next.

**Comment by netnazgul** [comment]
Yes, Bain's is a corner case when the only sound output is [briefly] disconnected during OpenRA operation, I didn't think it warranted a separate issue raised.

**Comment by pchote** [comment]
It turns out that this is not currently feasible without a serious rewrite of the sound code.
The current setup can't change devices on the fly - only on game start.
